AlphaWallet/Tokens/ViewControllers/TokenInstanceViewController.swift

```diff
@@ -142,10 +143,40 @@ class TokenInstanceViewController: UIViewController, TokenVerifiableStatusViewCo
                 button.isEnabled = false
             }
         }
+        buttonsBar.optionButtons.first?.addTarget(self, action: #selector(optionsButtonTapped), for: .touchUpInside)
 
         tokenRowView.configure(tokenHolder: tokenHolder, tokenView: .view, areDetailsVisible: tokenHolder.areDetailsVisible, width: 0, assetDefinitionStore: assetDefinitionStore)
     }
 
+    @objc private func optionsButtonTapped(sender: UIButton) {
```

Comment:
- [ ] Is it possible to extract this from `TokenInstanceViewController` and `TokensCardViewController` so that most of it (especially the actionsheet creation) is in the `ButtonsBar` class? Maybe add a delegate (protocol) to ButtonsBar with a single function which those 2 classes implement so they can call `handle(action:)`?

Then this line will not be needed:

```
buttonsBar.optionButtons.first?.addTarget(self, action: #selector(optionsButtonTapped), for: .touchUpInside)
```

Reply:
@hboon i guess we also need here dataSource, if we want to build UIAlertController inside ButtonsBar